File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int mj = nums[0];
        int cnt = 1;

        for(int i : nums){
            if(i == mj) cnt++;
            else if(i != mj) cnt--;
            if(cnt==0){
                cnt=1;
                mj=i;
            }
            
        }

        return mj;
    }
};

File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        n=len(nums)

        maj=nums[0]
        cnt=1

        for i in range(1,n):
            if(nums[i]==maj):
                cnt+=1
            if(nums[i]!=maj):
                cnt-=1
            if(cnt==0):
                cnt+=1
                maj=nums[i]
        
        return maj